gemrb/core/PathFinder.cpp

```diff
@@ -289,31 +295,36 @@ Path Map::FindPath(const Point& s, const Point& d, unsigned int size, unsigned i
 	// begin algo init
 	distFromStart[smptSource.y * mapSize.w + smptSource.x] = 0;
 	parents[smptSource.y * mapSize.w + smptSource.x] = nmptSource;
-	open.emplace(PQNode(nmptSource, 0));
+
+	openCost.push_back(0);
+	openPoint.emplace_back(nmptSource);
+
 	bool foundPath = false;
 	unsigned int squaredMinDist = minDistance * minDistance;
 
 	// Weighted heuristic. Finds sub-optimal paths but should be quite a bit faster
 	constexpr float_t HEURISTIC_WEIGHT = 1.5;
-	auto getHeuristic = [&](const SearchmapPoint& smptChild, const int& smptChildIdx) {
+	const auto getHeuristic = [&](const SearchmapPoint& smptChild, const int& smptChildIdx) {
 		// Calculate heuristic
-		int xDist = smptChild.x - smptDest.x;
-		int yDist = smptChild.y - smptDest.y;
+		const int xDist = smptChild.x - smptDest.x;
+		const int yDist = smptChild.y - smptDest.y;
 		// Tie-breaking used to smooth out the path
-		int dxCross = smptDest.x - smptSource.x;
-		int dyCross = smptDest.y - smptSource.y;
-		int crossProduct = std::abs(xDist * dyCross - yDist * dxCross) >> 3;
-		double distance = std::hypot(xDist, yDist);
-		double heuristic = HEURISTIC_WEIGHT * (distance + crossProduct);
-		double estDist = distFromStart[smptChildIdx] + heuristic;
+		const int dxCross = smptDest.x - smptSource.x;
+		const int dyCross = smptDest.y - smptSource.y;
+		const int crossProduct = std::abs(xDist * dyCross - yDist * dxCross) >> 3;
+		const float distance = std::hypot(xDist, yDist);
```

Comment:
Doesn't this cause narrowing by first casting to double? Using `std::hypotf` should avoid the final cast.

Reply:
It was more of a linter comment than a performance one.

Reply:
Yes, I know I know, but when you mentioned it, I was curious if the implementation differs for double precision. And at a first glance it seems like it does. This improvement is on you! ^^

Reply:
That's the shortcoming of looking at the summary of a summary, you loose basically all the context.
The case in which we had previously the highest median improvement percent was the case1 on bench1, which involves very high count of random short paths generated by NPCs. It is impossible to repeat this test _exactly_, so I wouldn't be concerned about this at all.
Funnily enough, when working on traversability cache, I got in this case1 scenario on bench0 such an amazing result with clearly not optimal solution, that I couldn't beat it even with the final solution, even though every other metric showed that final solution should outperform that interim version by a lot.